### ISSUE TITLE
sigmoid_focal_loss不支持ignore_index参数

### ISSUE THREAD

**Issue by lixuefeng810** [PaddlePaddle/Paddle#28068]
- 版本、环境信息
1）PaddlePaddle版本：1.8
2）CPU/GPU：不涉及
3）系统环境：Ubuntu 16.04

函数原型
paddle.fluid.layers.sigmoid_focal_loss(x, label, fg_num, gamma=2.0, alpha=0.25)
函数文档
https://www.paddlepaddle.org.cn/documentation/docs/zh/api_cn/layers_cn/sigmoid_focal_loss_cn.html

focal_loss不支持设置ignore_label, 进行图像增强的时候会在空白处填充固定像素值,而在计算loss时需要设置ignore该像素值,而在使用过focal loss时不支持设置ignore_index,使用起来不太方便

**Comment by FlyingQianMM** [comment]
您好，paddle 2.0中的sigmoid_focal_loss主要使用sigmoid与binary_cross_entropy_with_logits两个op组合实现，参考源码https://github.com/PaddlePaddle/Paddle/blob/a0b2f936895abfa633aeff46d621d337d79677d4/python/paddle/nn/functional/loss.py#L1278 。您可以参考这种实现方式，自行组合focal loss，通过binary_cross_entropy_with_logits的weights控制需要ignore的像素值。
